cmd/kubelet/app/server.go

```diff
@@ -488,6 +446,35 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		}
 	}
 
+	// Register current configuration with /configz endpoint
```

Comment:
Dynamic config needs to be fired up before we generate a KubeDeps, because these dependencies are generated based on the intended configuration. If remote configuration exists, and you weren't handed a KubeDeps, you want to use it to build a KubeDeps. Please move this back to where it was (before `if kubeDeps == nil`) :-).

Also: If a KubeDeps already exists, and it happens to have a client on it already, you can feel free to use that client, but be sure to check for the presence of that client and generate one if it does not exist.

Finally, I'm about to do a lot of major work and refactoring on this feature. Since a lot of things will change, it might not be the best use of your time to focus on this part of the Kubelet.